Northstar.Client/mod/scripts/vscripts/ui/menu_ns_serverbrowser.nut

```diff
@@ -1274,10 +1185,120 @@ void function RemoveConnectToServerCallback( void functionref( ServerInfo ) call
 	file.connectCallbacks.fastremovebyvalue( callback )
 }
 
-void function TriggerConnectToServerCallbacks()
+void function TriggerConnectToServerCallbacks( ServerInfo ornull targetServer = null )
 {
+	ServerInfo server
+	if (targetServer == null)
+	{
+		targetServer = file.lastSelectedServer
+	}
+
```

Comment:
```suggestion

expect ServerInfo( targetServer )
```
Not sure how badly `expect` hurts performance, but there should be no need to call it for every callback